components/script/webdriver_handlers.rs

```diff
@@ -185,6 +185,44 @@ pub fn handle_get_title(page: &Rc<Page>, _pipeline: PipelineId, reply: IpcSender
     reply.send(String::from(page.document().Title())).unwrap();
 }
 
+pub fn handle_get_rect(page: &Rc<Page>,
+                       pipeline: PipelineId,
+                       element_id: String,
+                       reply: IpcSender<Result<WebDriverElementRect, ()>>) {
+    reply.send(match find_element_by_element_id(&*page, pipeline, element_id) {
+        Some(elem) => {
+            // https://w3c.github.io/webdriver/webdriver-spec.html#dfn-calculate-the-absolute-position
+            match elem.downcast::<HTMLElement>() {
+                Some(html_elem) => {
+                    // Step 1
+                    let mut x: i32 = 0;
+                    let mut y: i32 = 0;
+
+                    let mut offset_parent = html_elem.GetOffsetParent();
+
+                    // Step 2
+                    while let Some(element) = offset_parent {
+                        offset_parent = element.downcast::<HTMLElement>().and_then(|elem| {
+                            x += elem.OffsetLeft();
+                            y += elem.OffsetTop();
```

Comment:
- Is there documentation for this?

  - Is this the case for all iterator adapters or just the `Option<T>` methods that take a `FnOnce` or just `and_then`?

Ideally I'd like to create a `OffsetParentIterator` struct, impl `Iterator`, and use `fold`, but this seems to be the only case where this would be used. Without an iterator as far as I know we're pretty much stuck to a `while` loop and mutating `x` and `y`. Let me know if I'm missing anything. Thanks for the comments!